packages/common/src/ArcGISContext.ts

```diff
@@ -280,6 +302,47 @@ export class ArcGISContext implements IArcGISContext {
     if (opts.properties) {
       this._properties = opts.properties;
     }
+
+    if (opts.environment) {
+      this._environment = opts.environment;
+    } else {
+      let environment: HubEnvironment = "production";
+      const portalSubdomain =
+        this._portalSelf?.portalHostname.split(".")[0] ||
+        getPortalBaseFromOrgUrl(this._portalUrl);
+      const portalSubdomainHash = {
+        devext: "devext",
+        qaext: "qaext",
+        www: "production",
+        // enterprise: "enterprise",
+        // sandbox: "sandbox",
+        // localdev: "localdev"
+      };
+      environment = (portalSubdomainHash as any)[portalSubdomain];
+      if (this._portalSelf.isPortal) {
+        environment = "enterprise";
+      }
+      this._environment = environment;
+    }
+
+    const location = _getLocation();
+    if (opts.features) {
+      this._features = opts.features;
+    } else if (location) {
+      const params = new URLSearchParams(location.search);
+      const paramName = "hubFeatures";
+      if (params.has(paramName)) {
+        this._features = params
+          .getAll(paramName)
+          .reduce((hubFeatures, feature) => {
+            const [key, val] = feature.split(":");
+            if (key && val !== undefined) {
+              hubFeatures[key.trim()] = val.trim() === "true";
+            }
+            return hubFeatures;
+          }, {} as IHubFeatures);
+      }
+    }
```

Comment:
Rough implementation: If we are passed feature overrides, use them, if not, get them from the URL qquery params.